### PR TITLE
Add usable dockerfile to clightning

### PR DESCRIPTION
**Pull request by NicolasDorier**
@cdecker, you are providing a dockerfile on https://hub.docker.com/r/cdecker/lightningd .
Sadly, this docker file is useless for developers as it is impossible to connect to the RPC unix domain socket.

This PR propose a `Dockerfile` which expose clightning on TCP with `socat`. This allow developers not using linux to still develop stuff on top of CLightning by dockerizing the app and connecting via TCP.

It will also simplify `lightning-charge` which currently is starting clightning and charge in the same container though a very hacky manner.

This port is not meant to be mapped to the host (except for testing) for obvious security reasons.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Example of dockerfile using it:

```
version: "3"

services:
  bitcoind:
    image: nicolasdorier/docker-bitcoin:0.16.0
    environment:
      BITCOIN_EXTRA_ARGS: |
        rpcuser=ceiwHEbqWI83
        rpcpassword=DwubwWsoo3
        regtest=1
        server=1
        rpcport=43782
        port=39388
        whitelist=0.0.0.0/0
        zmqpubrawblock=tcp://0.0.0.0:29000
        zmqpubrawtx=tcp://0.0.0.0:29000
        txindex=1
        # Eclair is still using addwitnessaddress
        deprecatedrpc=addwitnessaddress 
    expose:
      - "43782" # RPC
      - "39388" # P2P
    volumes:
      - "lightning_datadir:/data"

  lightning:
    image: nicolasdorier/clightning
    environment:
      LIGHTNINGD_OPT: |
        network=regtest
        bitcoin-rpcconnect=bitcoind
    volumes:
      - "lightning_datadir:/root/.bitcoin"
    expose:
      - "9835" # RPC
      - "9735" # Lightning
    links:
      - bitcoind

volumes:
  lightning_datadir:
```

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Declare files that will always have CRLF line endings on checkout.
+*.sh text eol=lf
+*.py text eol=lf
+Makefile text eol=lf
```
Comment (rustyrussell) [inline .gitattributes]: I'm confused.

Is this for Windows or something?  Why not just check them in with the correct formatting?

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Declare files that will always have CRLF line endings on checkout.
```
Comment (cdecker) [inline .gitattributes]: This is a duplicate commit from #1185

**Comment by hbasria** [comment]
If you want to use as a rest api  

pip install lightning-rest
python -m lightning_rest.server 8000 ~/.lightning/lightning-rpc

or

docker run -it --rm -p 8000:8000 -v /path/to/lightning-rpc:/tmp/lightning-rpc siriuslabs/lightning-rest 8000 /tmp/lightning-rpc

https://github.com/hbasria/lightning-rest

**Comment by NicolasDorier** [comment]
> lightningd does not have any authentication simply because we want higher up layers to implement whatever the environment it is running in requires.

This is exactly what I want to do... except that it is impossible without an hell of pain because I am on windows. I can't just code in visual studio while connecting to clightning.

> If you want to use as a rest api

This is cool. Though my use case is to use TCP only for dev purposes, this goes away in prod by using shared volumes as you are doing.

**Comment by NicolasDorier** [comment]
closing this, will maintain my own `nicolasdorier/clightning` and make things depend on this instead.

**Pull request closed without merging**